pkg/pipeline/extract/extract_timebased.go

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (C) 2022 IBM, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package extract
+
+import (
+	"time"
+
+	"github.com/netobserv/flowlogs-pipeline/pkg/api"
+	"github.com/netobserv/flowlogs-pipeline/pkg/config"
+	"github.com/netobserv/flowlogs-pipeline/pkg/pipeline/extract/timebased"
+	log "github.com/sirupsen/logrus"
+)
+
+type ExtractTimebased struct {
+	Filters          []timebased.FilterStruct
+	RecordKeyStructs map[string]*timebased.RecordKeyTable
+}
+
+// Extract extracts a flow before being stored
+func (et *ExtractTimebased) Extract(entries []config.GenericMap) []config.GenericMap {
+	log.Debugf("entering ExtractTimebased Extract")
+	nowInSecs := time.Now().Unix()
+	// Populate the Table with the current entries
+	for _, entry := range entries {
+		log.Debugf("ExtractTimebased Extract, entry = %v", entry)
+		timebased.AddEntryToTables(et.RecordKeyStructs, entry, nowInSecs)
+	}
+
+	output := make([]config.GenericMap, 0)
+	// Calculate Filters based on time windows
+	for _, filter := range et.Filters {
+		filter.CalculateResults(nowInSecs)
+		filter.ComputeTopkBotk()
+		genMap := filter.CreateGenericMap()
+		output = append(output, genMap...)
+	}
```

Comment:
If `CalculateResults()` and `ComputeTopkBotk()` are executed on every incoming batch of flowLogs, this might be cpu intensive when the frequency of incoming batches is high.

Reply:
Let's discuss what is the desired frequency to compute. Should we add another configuration parameter?